.github/workflows/nightly.yml

```diff
@@ -0,0 +1,24 @@
+name: 'Nightly'
+
+on:
+  schedule:
+    - cron: '* 8 * * *' # 8am in UTC is midnight PST
+    branches:
+      - master
+
+jobs:
+  test-all-deprecations:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12.x
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install dependencies for master
+        run: yarn install
+      - name: Basic Tests
+        env:
+          CI: true
+          ASSERT_ALL_DEPRECATIONS: true
+        run: yarn test
```

Comment:
we should run

- basic tests
- fastboot tests
- fetch tests
- encapsulation tests
- canary/beta Ember tests

Reply:
In a future PR we should also consider how to monitor for deprecations in the relationship-performance-test-app, but that can be ignored for now.

Reply:
I didn't set up fastboot tests and they aren't working right now, so not adding them currently. What are the fetch tests?
Added the others.

Reply:
@Gaurav0 can you clarify what you mean by:

> I didn't set up fastboot tests and they aren't working right now

I am happy to merge this and add fastboot and fetch tests in follow up

Reply:
Yes, I reverted updating test-helper.js in fastboot-test-app to assert on deprecations. I think the code I wrote with the custom QUnitAdapter for testem doesn't work in fastboot, that may be why it fails.